### PR TITLE
feat: Gidk-#2hgfrf | Prateek | Stage | Multi-currency support in Voucher adjustment (Ledger, voucher module, invoice preview)

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
https://app.clickup.com/t/2hgfrf

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 1]
```diff
@@ -1603,9 +1603,30 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
...
             }
             item['grandTotalTooltipText'] = `In ${this.baseCurrency}: ${grandTotalAmountForCompany}<br />(Conversion Rate: ${grandTotalConversionRate})`;
-            item['balanceDueTooltipText'] = `In ${this.baseCurrency}: ${balanceDueAmountForCompany}<br />(Conversion Rate: ${balanceDueAmountConversionRate})`;
+            if (item.gainLoss) {
+                item['balanceDueTooltipText'] = `In ${this.baseCurrency}: ${balanceDueAmountForCompany}<br />(Exchange ${item.gainLoss > 0 ? 'gain' : 'loss'}: ${Math.abs(item.gainLoss)})`;
+            } else {
+                item['balanceDueTooltipText'] = `In ${this.baseCurrency}: ${balanceDueAmountForCompany}<br />(Conversion Rate: ${balanceDueAmountConversionRate})`;
+            }
+
+            // let text = this.localeData?.currency_conversion;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts:1615]: WIll uncomment after multi-lingual v2 changes are deployed on STAGE

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html [hunk 2]
```diff
@@ -523,13 +522,15 @@
...
 
                                         <!-- added multicurrency feature make it functional to change into appropriate currency -->
-                                        <span class="sp-rupees d-flex justify-content-end"
-                                            *ngIf="isMulticurrencyAccount">
-                                            <amount-field
-                                                [amount]="invFormData.voucherDetails.balanceDue * exchangeRate"
-                                                [currencySymbol]="baseCurrencySymbol" [currencyCode]="companyCurrency">
-                                            </amount-field>
-                                        </span>
+                                            <span *ngIf="isSalesInvoice && isMulticurrencyAccount && adjustPaymentBalanceDueData === 0 && invFormData.voucherDetails.balanceDue === 0" class="balance-due">Exchange {{getConvertedBalanceDue() <= 0 ? 'gain' : 'loss'}}:&nbsp;</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html:525]: Delete this

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html [hunk 3]
```diff
@@ -81,12 +81,11 @@
                                             </span>
 
                                             <!-- added multicurrency feature make it functional to change into appropriate currency -->
-
-
+                                            <span *ngIf="isSalesInvoice && isMulticurrencyAccount && adjustPaymentBalanceDueData === 0 && invFormData.voucherDetails.balanceDue === 0" class="balance-due">Exchange {{getConvertedBalanceDue() <= 0 ? 'gain' : 'loss'}}:&nbsp;</span>
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html:84]: Delete this

File: apps/web-giddh/src/assets/locale/invoice/en.json [hunk 4]
```diff
@@ -127,6 +127,17 @@
     "search_particular": "Search Particular",
     "search_description": "Search Description",
     "currency_conversion": "In [BASE_CURRENCY]: [AMOUNT]<br />(Conversion Rate: [CONVERSION_RATE])",
+    "create_e_invoice": "Create E-invoice",
+    "e_invoice_status": "E-invoice Status",
+    "e_invoice_statuses": {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/invoice/en.json:132]: duplicate key added

File: apps/web-giddh/src/assets/locale/invoice/en.json [hunk 5]
```diff
@@ -127,6 +127,17 @@
     "search_particular": "Search Particular",
     "search_description": "Search Description",
     "currency_conversion": "In [BASE_CURRENCY]: [AMOUNT]<br />(Conversion Rate: [CONVERSION_RATE])",
+    "create_e_invoice": "Create E-invoice",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/invoice/en.json:130]: duplicate key added

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**